plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployJobRunner.java

```diff
@@ -0,0 +1,25 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+
+/**
+ * Runs a {@link AppEngineDeployJob} in an {@link IWorkspace} using a lock to prevent any changes to the project while
+ * the operation is in progress.
+ *
+ */
+// Main purpose of this class is to separate Eclipse' job execution from other logic to enhance testability
+public class DeployJobRunner {
+
+  public void runJob(AppEngineDeployJob deployJob, IProject project, IProgressMonitor monitor) throws CoreException {
```

Comment:
Removed `AppEngineDeployJob` marker interface, and replaced it with Eclipse's `IWorkspaceRunnable`, renamed the class to indicate its more general applicability.